src/com/stuypulse/stuylib/math/interpolation/IntervalInterpolator.java

```diff
@@ -0,0 +1,37 @@
+/* Copyright (c) 2022 StuyPulse Robotics. All rights reserved. */
+/* This work is licensed under the terms of the MIT license */
+/* found in the root directory of this project. */
+package com.stuypulse.stuylib.math.interpolation;
+
+import com.stuypulse.stuylib.math.Vector2D;
+
+
+/**
+ * This class uses two reference points to interpolate for a third point's y value
```

Comment:
My suggestion: This class uses two reference points to interpolate all points on the line between them.

Reply:
maybe specify the line is linear?